ksqldb-engine/src/main/java/io/confluent/ksql/logicalplanner/LogicalPlanner.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.logicalplanner;
+
+import static io.confluent.ksql.metastore.model.DataSource.DataSourceType;
+
+import com.google.common.collect.ImmutableSet;
+import io.confluent.ksql.logicalplanner.nodes.SelectNode;
+import io.confluent.ksql.logicalplanner.nodes.SourceNode;
+import io.confluent.ksql.logicalplanner.nodes.StreamSourceNode;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.metastore.model.DataSource;
+import io.confluent.ksql.name.SourceName;
+import io.confluent.ksql.parser.tree.AliasedRelation;
+import io.confluent.ksql.parser.tree.Query;
+import io.confluent.ksql.parser.tree.Select;
+import java.util.Objects;
+
+public final class LogicalPlanner {
+
+  private LogicalPlanner() {}
+
+  public static LogicalPlan buildLogicalPlan(
+      final MetaStore metaStore,
+      final Query query
+  ) {
+    Objects.requireNonNull(metaStore, "metaStore");
+    Objects.requireNonNull(query, "query");
+
+    // cast will fail as long as we don't support joins
+    // that's ok for now as we check this condition upfront
+    final AliasedRelation inputStreamOrTable = (AliasedRelation) query.getFrom();
+
+    final SourceName sourceName = inputStreamOrTable.getAlias();
+    final DataSource sourceStreamOrTable = metaStore.getSource(sourceName);
+
+    final SourceNode<?> sourceNode;
+    if (sourceStreamOrTable.getDataSourceType() == DataSourceType.KSTREAM) {
+      sourceNode = new StreamSourceNode(sourceName, sourceStreamOrTable.getSchema());
+    } else if (sourceStreamOrTable.getDataSourceType() == DataSourceType.KTABLE) {
+      throw new UnsupportedOperationException("Input TABLES are not supported by the new planner");
+      //sourceNode = new TableSourceNode(sourceName, sourceStreamOrTable.getSchema());
```

Comment:
Are you leaving comments like this to show future strategies?

Reply:
Yes. Did not try to extend the scope to add input `TABLE` support yet. It's the intended code when we add `TABLES`.